spec/lib/artifact.spec.js

```diff
@@ -0,0 +1,292 @@
+                return {
+                    form: () => {
+                        return {
+                            append: () => {
+                                return resolver(null, {
+                                    headers: {
+                                        location: "someLocation"
+                                    }
+                                });
+                            }
+                        };
+                    }
+                };
+            });
+
+            return instance.upload(content).then(() => {
+                expect(requestMock.post).toHaveBeenCalled();
+            });
+        });
+        it("calls responseHandler.handleErrorResponse on error", () => {
+            responseHandlerMock.isErrorCode.andReturn(true);
+            spyOn(bluebird, "fromCallback").andReturn(bluebird.resolve({
+                headers: {
+                    location: "someLocation"
+                }
+            }));
+
+            return instance.upload(content).then(() => {
+                expect(responseHandlerMock.handleErrorResponse).toHaveBeenCalled();
+            });
+        });
+        it("calls responseHandler.resolveLink", () => {
+            spyOn(bluebird, "fromCallback").andReturn(bluebird.resolve({
+                headers: {
+                    location: "someLocation"
+                }
+            }));
+
+            return instance.upload(content).then(() => {
+                expect(responseHandlerMock.resolveLink).toHaveBeenCalled();
+            });
+        });
+    });
+    describe(".uploadFromFile()", () => {
+        beforeEach(() => {
+            responseHandlerMock.isErrorCode.andReturn(true);
+        });
+        it("calls bluebird.fromCallback()", () => {
+            spyOn(bluebird, "fromCallback").andReturn(bluebird.resolve({
+                headers: {
+                    location: "someLocation"
+                }
+            }));
+
+            return instance.uploadFromFile(content).then(() => {
+                expect(bluebird.fromCallback).toHaveBeenCalled();
+            });
+        });
+        it("creates a read stream if the file is a string", () => {
+            spyOn(fs, "createReadStream");
+            requestMock.post.andCallFake((options, resolver) => {
+                resolver(null, {
+                    statusCode: "exampleStatus",
+                    headers: {
+                        location: "exampleLocation"
+                    }
+                });
+            });
+
+            return instance.uploadFromFile(content).then(() => {
+                expect(fs.createReadStream).toHaveBeenCalled();
+                expect(requestOptionsMock.createOptions).toHaveBeenCalled();
+                expect(requestMock.post).toHaveBeenCalled();
+            });
+        });
+        it("doesn't create a read stream if the file isn't a string", () => {
+            spyOn(fs, "createReadStream");
```

Comment:
This test and the previous could go inside a describe block since you make spyOn(fs, "createReadStream") twice. Just an idea.